recipe-server/client/control/state/recipes/reducers.js

```diff
@@ -31,9 +31,25 @@ function history(state = new Map(), action) {
 
 
 function items(state = new Map(), action) {
+  let recipe;
+
   switch (action.type) {
     case RECIPE_RECEIVE:
-      return state.set(action.recipe.id, fromJS(action.recipe));
+      recipe = fromJS(action.recipe);
+      recipe = recipe.set('action_id', action.recipe.action.id);
+      recipe = recipe.remove('action');
+
+      recipe = recipe.set('latest_revision_id', action.recipe.latest_revision.id);
+      recipe = recipe.remove('latest_revision');
+
+      if (action.recipe.approved_revision) {
+        recipe = recipe.set('approved_revision_id', action.recipe.approved_revision.id);
+      } else {
+        recipe = recipe.set('approved_revision_id', null);
```

Comment:
Generally, I put the `else` stuff as the default and then leave only the `if`, e.g.:

```
recipe = recipe.set('approved_revision_id', null);

if (action.recipe.approved_revision) {
          recipe = recipe.set('approved_revision_id', action.recipe.approved_revision.id);
}
```